dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/impl/NettyFrameworkImpl.java

```diff
@@ -0,0 +1,359 @@
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.ffdc.annotation.FFDCIgnore;
+import com.ibm.ws.kernel.feature.ServerStarted;
+import com.ibm.wsspi.kernel.service.utils.ServerQuiesceListener;
+
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioDatagramChannel;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.util.concurrent.Future;
+import io.openliberty.netty.internal.BootstrapExtended;
+import io.openliberty.netty.internal.NettyFramework;
+import io.openliberty.netty.internal.ServerBootstrapExtended;
+import io.openliberty.netty.internal.exception.NettyException;
+
+/**
+ * Liberty NettyFramework implementation bundle
+ */
+@Component(configurationPid = "io.openliberty.netty.internal",
+    immediate = true, 
+    service = { NettyFramework.class, ServerQuiesceListener.class }, 
+    property = { "service.vendor=IBM" })
+public class NettyFrameworkImpl implements ServerQuiesceListener, NettyFramework {
+
+    private static final TraceComponent tc = Tr.register(NettyFrameworkImpl.class, NettyMessageConstants.NETTY_TRACE_NAME, NettyMessageConstants.NETTY_BUNDLE);
+
+    /** Reference to the executor service -- required */
+    private ExecutorService executorService = null;
+
+    /** server started logic borrowed from CHFWBundle */
+    private static AtomicBoolean serverCompletelyStarted = new AtomicBoolean(false);
+    private static Queue<Callable<?>> serverStartedTasks = new LinkedBlockingQueue<>();
+    private static Object syncStarted = new Object() {
+    }; // use brackets/inner class to make lock appear in dumps using class name
+    Set<Channel> activeChannels = ConcurrentHashMap.newKeySet();
+
+    protected EventLoopGroup parentGroup;
+    protected EventLoopGroup childGroup;
+    
+    private CHFWBundle chfw;
+    private volatile boolean isActive = false;
+
+    
+    /**
+     * DS method for setting the required channel framework service.
+     * TODO: for now this reference is needed for access to EndPointMgr. That code will be split out.
+     *
+     * @param bundle
+     */
+    @Reference(name = "chfwBundle")
+    protected void setChfwBundle(CHFWBundle bundle) {
+        chfw = bundle;
+    }
+
+    /**
+     * This is a required static reference, this won't
+     * be called until the component has been deactivated
+     *
+     * @param bundle CHFWBundle instance to unset
+     */
+    protected void unsetChfwBundle(CHFWBundle bundle) {}
+
+    @Activate
+    protected void activate(ComponentContext context, Map<String, Object> config) {
+        // use the executor service provided by Liberty
+        parentGroup = new NioEventLoopGroup(1, executorService);
+        childGroup = new NioEventLoopGroup(0, executorService);
```

Comment:
Is it worth a comment that 0 for the threads uses the 'default' number of threads - and is is OK for the external library to have control over this when we have 'io.openliberty.io.netty;version=latest' in the bnd file (so I guess it could change if netty version changes?)

Reply:
I've opened an issue to revisit this https://github.com/OpenLiberty/open-liberty/issues/18515